src/core/observers/field_observer.ts

```diff
@@ -0,0 +1,44 @@
+import { FieldController } from '../field_controller';
+import { ValidationStrategy } from '../validation_strategy';
+import { BlurObserver, InputObserver, SubmitObserver } from './field';
+
+export class FieldObserver {
+  readonly field: FieldController;
+  readonly blurObserver: BlurObserver;
+  readonly inputObserver: InputObserver;
+  readonly submitObserver: SubmitObserver;
+
+  private started: boolean
+
+  constructor(field: FieldController) {
+    this.field = field;
+    this.started = false;
+
+    this.blurObserver = new BlurObserver(field);
+    this.inputObserver = new InputObserver(field);
+    this.submitObserver = new SubmitObserver(field);
+  }
+
+  start() {
+    if (!this.started) {
+      if (this.field.validationStrategy === ValidationStrategy.Touch) {
+        this.blurObserver.start();
+        this.inputObserver.start();
+      }
+
+      this.submitObserver.start();
```

Comment:
Se a gente for seguir o exemplo do react-hook-form, depois que o form é enviado uma vez, os campos são validados depois de todo input, só que do jeito que está aqui não rolaria como está.